nautobot/utilities/views.py

```diff
@@ -151,3 +187,1051 @@ def get_return_url(self, request, obj=None):
 
         # If all else fails, return home. Ideally this should never happen.
         return reverse("home")
+
+
+class NautobotViewSetMixin:
+
+    routes = []
+    model = None
+    queryset = None
+    table = None
+    template_name = None
+    filterset = None
+    filterset_form = None
+    import_form = None
+    prefetch_related = []
+
+    def detail_queryset(self):
+        return self.model.objects.all()
+
+    def table_queryset(self):
+        return self.model.objects.prefetch_related(*self.prefetch_related)
+
+    def define_routes(self):
+        return []
+
+    def get_extra_context(self, request, view_type, instance=None):
+        """
+        Return any additional context data for the template.
+        request: The current request
+        instance: The object being viewed
+        """
+        return {}
+
+    def get_template_name(self, view_type):
+        # Use "<app>/<model>_<view_type> if available, else fall back to generic templates
+        model_opts = self.model._meta
+        app_label = model_opts.app_label
+        if view_type == "detail":
+            return f"{app_label}/{model_opts.model_name}.html"
+
+        try:
+            select_template([f"{app_label}/{model_opts.model_name}_{view_type}.html"])
+            return f"{app_label}/{model_opts.model_name}_{view_type}.html"
+        except TemplateDoesNotExist:
+            return f"generic/object_{view_type}.html"
+
+
+class ObjectDetailViewMixin(NautobotViewSetMixin):
```

Comment:
I'm concerned that none of these other mixing are being derived from existing mixins found in `rest_framework.mixins` because each of these actions already have corresponding view methods that we do not appear to be using at all (e.g. `list()`, `retrieve()`, `perform_update()`,  `partial_update()`, etc.)

In my opinion those would be better to align around because they are existing and also don't require proliferation of all of these different custom `handle_{thing}_for_{action}` methods.